### PR TITLE
Added PThreadPool v0.0.20191029

### PR DESCRIPTION
**Pull request by stemann**
Matches version used by PyTorch v1.5.0

### CODE REVIEW

File: P/PThreadPool/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "PThreadPool"
+version = v"0.0.20191029"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/Maratyszcza/pthreadpool.git", "d465747660ecf9ebbaddf8c3db37e4a13d0c9103"),
+    GitSource("https://github.com/Maratyszcza/FXdiv.git", "b742d1143724d646cd0f914646f1240eacf5bd73"),
```
Comment (giordano) [inline P/PThreadPool/build_tarballs.jl:11]: Can't this be its own package?

File: P/PThreadPool/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd pthreadpool
```
Comment (giordano) [inline P/PThreadPool/build_tarballs.jl:17]:
```suggestion
cd $WORKSPACE/srcdir/pthreadpool
```

File: P/PThreadPool/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    ..
+cmake --build . -- -j $nproc
+make install
+install_license ../LICENSE
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+filter!(p -> !Sys.iswindows(p), platforms) # Windows fails to build
```
Comment (giordano) [inline P/PThreadPool/build_tarballs.jl:37]: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=17902&view=logs&j=bf08fc6b-a6d6-514d-d910-d67f80a9f009&t=d1c794c2-60c3-5b4e-76f3-47455db146a8&l=232

File: P/PThreadPool/build_tarballs.jl (hunk 1 above)
Comment (stemann) [reply P/PThreadPool/build_tarballs.jl:11]: It's header-only, so I guess not?

Comment (giordano) [reply P/PThreadPool/build_tarballs.jl:11]: Why not?

File: P/PThreadPool/build_tarballs.jl (hunk 3 above)
Comment (stemann) [reply P/PThreadPool/build_tarballs.jl:36]: It's the same reason as for filtering out a few platforms for CPUInfo: I'm trying to match the exact version used by PyTorch v1.5.0, cf. https://github.com/pytorch/pytorch/tree/v1.5.0/third_party - even if a few (major) platforms don't build (out of the box).

File: P/PThreadPool/build_tarballs.jl (hunk 1 above)
Comment (stemann) [reply P/PThreadPool/build_tarballs.jl:11]: Ah yes - of course it can...

Comment (giordano) [reply P/PThreadPool/build_tarballs.jl]: Uhm, it appears they need to have the full source code of the dependency including the cmake file. OK, I'll refrain from making any further comments on what they do here.

Comment (stemann) [reply P/PThreadPool/build_tarballs.jl:11]: Yeah - I saw that too.

I've left FXdiv as a BuildDependency even though it isn't actually

Comment (stemann) [reply P/PThreadPool/build_tarballs.jl:11]: ...  - at least it serves to document FXdiv as a dependency in the Pkg-system.

Comment (stemann) [reply P/PThreadPool/build_tarballs.jl:11]: err... - it's just a BuildDependency, so: Keep it or remove it?

Comment (giordano) [reply P/PThreadPool/build_tarballs.jl:11]: Remove it

Comment (stemann) [reply P/PThreadPool/build_tarballs.jl:11]: Done

File: P/PThreadPool/build_tarballs.jl (hunk 3 above)
Comment (stemann) [reply P/PThreadPool/build_tarballs.jl:36]: It was this commit: https://github.com/Maratyszcza/pthreadpool/commit/ebd50d0cfa3664d454ffdf246fcd228c3b370a11 (the commit following the 2019-10-29 version: https://github.com/Maratyszcza/pthreadpool/commit/d465747660ecf9ebbaddf8c3db37e4a13d0c9103)

Comment (giordano) [reply P/PThreadPool/build_tarballs.jl:36]: How about using https://github.com/Maratyszcza/pthreadpool/commit/ebd50d0cfa3664d454ffdf246fcd228c3b370a11 then?

Comment (stemann) [reply P/PThreadPool/build_tarballs.jl:36]: Will add that with #4566

**Pull request merged**